engine/engine.go

```diff
@@ -319,7 +331,10 @@ func (e *Engine) sendDHMetadata(ctx context.Context, putMetaReq dhstore.PutMetad
 	if err != nil {
 		return err
 	}
+
 	req.Header.Set("Content-Type", "application/json")
+	shardKey := base64.RawStdEncoding.EncodeToString(putMetaReq.Key)
```

Comment:
We discussed `base58` on slack. This doesn't matter as long as it's consistent with read side header. Cc @ischasny

Reply:
> I am curious why re-encode the shard key to base64 at all instead of using the base58 URL suffix straight up?

+1 to that - lets use base58 as we use in URLs and serve in payloads.

Reply:
I chose to use base64 as the encoding is faster and part of the stdlib, and no need to use base58. We can always use base58, but I prefer base64 unless there is a reason no to.

Reply:
Well I guess the main reason is consistency as all urls have base58 encoded multihashes. If we use base64 then we'd have to convert base58 -> base64 on read anyway, that would defeat any potential performance boost from using base64 in the first place.

Reply:
I was thinking that the shard key is not a URL, never goes into a URL, and is only used as a header value (consider it like a cookie).

... but on the Read size this would result in a need for two different encodings:
- base64 for the shard key
- base58 for find URL

Is that correct?  If so, then I will change to base58.

Reply:
Yes, this is what I mean. URL already encoded as base58...

Reply:
- Consistency in calculating the shard key is all that’s required.
- The choice between base58 and base64 is only relevant here as an optimsation, where it saves extra re-encoding of the key since (i believe) in majority of the paths the key is already base58 encoded.
- Aside from the above I see no benefit or disadvantage of using one over the other unless there is something that I have missed.
- HTTP header value is typically considered to be case-sensitive. It’s the key [that’s case-insensitive](https://www.rfc-editor.org/rfc/rfc7230#section-3.2)
- To be on the safe side maybe it makes sense to use an encoding that would work regardless of whether the value is case-sensitive or not.

Reply:
changed to base58